src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/PluginIT.java

```diff
@@ -0,0 +1,95 @@
+import static org.hamcrest.Matchers.equalTo;
+
+public class PluginIT extends SQLIntegTestCase {
+
+    @Override
+    protected void init() throws Exception {
+        loadIndex(Index.ACCOUNT);
+    }
+
+    @Test
+    public void sqlEnableSettingsTest() throws IOException {
+        updateClusterSettings(SettingType.PERSISTENT, "opendistro.sql.enabled", "true");
+        JSONObject queryResult = executeQuery(String.format("SELECT firstname FROM %s/account WHERE account_number=1", TEST_INDEX_ACCOUNT));
+        System.out.println(queryResult);
+        Assert.assertThat(getHits(queryResult).length(), equalTo(1));
+
+        updateClusterSettings(SettingType.PERSISTENT, "opendistro.sql.enabled", "false");
+        Response response = null;
+        try {
+            queryResult = executeQuery(String.format("SELECT firstname FROM %s/account WHERE account_number = 1", TEST_INDEX_ACCOUNT));
+            System.out.println(queryResult);
+        } catch (ResponseException ex) {
+            response = ex.getResponse();
+        }
+
+        queryResult = new JSONObject(TestUtils.getResponseBody(response));
+        Assert.assertThat(queryResult.getInt("status"), equalTo(400));
+        JSONObject error = queryResult.getJSONObject("error");
+        Assert.assertThat(error.getString("reason"), equalTo("Invalid SQL query"));
+        Assert.assertThat(error.getString("details"), equalTo("Either opendistro.sql.enabled or rest.action.multi.allow_explicit_index setting is false"));
+        Assert.assertThat(error.getString("type"), equalTo("SQLFeatureDisabledException"));
+        resetClusterSettings(SettingType.PERSISTENT, "opendistro.sql.enabled");
+    }
+
+    private JSONObject updateClusterSettings(SettingType settingType, String setting , String value) throws IOException {
+        Request request = new Request("PUT", "/_cluster/settings?pretty");
+        String persistentSetting = "{\""+ settingType.val() +"\": {\"" + setting + "\": " + value + "}}";
+        request.setJsonEntity(persistentSetting);
+        RequestOptions.Builder restOptionsBuilder = RequestOptions.DEFAULT.toBuilder();
+        restOptionsBuilder.addHeader("Content-Type", "application/json");
+        request.setOptions(restOptionsBuilder);
+        JSONObject result = new JSONObject(executeRequest(request));
+        System.out.println(result);
+        return result;
+    }
+
+    /**
+     * We need to reset all the cluster settings, otherwise ESIntegTestCase will throw
+     * java.lang.AssertionError: test leaves persistent cluster metadata behind
+     *
+     * TODO: Look for a way to reset all the persistent and transient settings instead of
+     *       individually resetting of each such setting
+     */
+    private JSONObject resetClusterSettings(SettingType settingType, String setting) throws IOException {
+        return updateClusterSettings(settingType, setting, "null");
+    }
+
+    private String executeRequest(final Request request) throws IOException {
```

Comment:
you could take the existing `SqlIntegTestCase::executeRequest(Request)` method (see [here](https://github.com/opendistro-for-elasticsearch/sql/blob/d0f2e0b5a72df8094a57af2250b45460ed92ba3a/src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLIntegTestCase.java#L157-L175)), and just change the assert line to compare against a provided status code instead of hardcoded `200`.

Reply:
We are not asserting anything here. Also its a private method that I can't use directly in `PluginIT`. I could make it protected and change the signature to `SqlIntegTestCase::executeRequest(Request, statusCode)` and change the all the usages, if that's what intended.

Reply:
ok, up to you, I just wanted to avoid much duplicate code.

Reply:
Fortunately , I was able to reuse `SqlIntegTestCase::executeRequest(Request)` without changing signature. But still needed `TestUtils.getResponseBody` to parse response from exception.